tests/next_tests/integration_tests/feature_tests/ffront_tests/test_execution.py

```diff
@@ -1108,7 +1108,7 @@ def unpack(
 def test_tuple_unpacking_too_many_values(cartesian_case):
```

Comment:
This test was previously shadowed by the succeeding one (now renamed), and turned out to fail. The changes to `foast_passes.type_deduction` are to make this one pass.